proto/mrt/mrt.c

```diff
@@ -551,8 +551,16 @@ mrt_rib_table_entry(struct mrt_table_dump_state *s, rte *r)
          * mp_next_hop is always encoded as ipv6 even when it's an ipv4
          * write mp_next_hop as an MP_REACH_NLRI field
          */
-        if ( 16 == next_hop->u.ptr->length ) {
+        if ( 16 == next_hop->u.ptr->length || 32 == next_hop->u.ptr->length) {
            ip6_addr *addr = (void *) s->bws->mp_next_hop->u.ptr->data;
+           if (32 == next_hop->u.ptr->length) {
```

Comment:
we should add some comment here that this is when the neighbor router is misbehaving - sending 2 nexthops is not really allowed by bgp

Reply:
Could this be a load balancing scheme?

Reply:
Should we perhaps also add some sort of warning log (or printf if we don't have a formal logger in the bird2 reepo) if the length is > 32? For example if for some reason we saw 3 addresses? I know we don't expect that to happen, but we also didn't expect 2 addresses to happen....

In fact we may even want to add a log/print when we see 2 addresses, also, if this is unexpected.

Reply:
actually I advise against this - there might be millions of routes advertised by faulty CISCO routers in this manner. Logging anything like that would be fairly useless from a diagnostic purpose. That logging would be turned off in production anyway because we would run into disk space. (At least that has been my experiencing)

Our current process of obtaining a tcpdump is less intrusive

Reply:
@zayfod  I bet my money on that this is a bug in a specific CISCO ios version and that the second link local address slips through - afaik BGP has no mechanism to select a better? nexthop. it is entirely based on the announcement of one nexthop. Other bugs that I have experience with BGP are

-  the announcement of an extended length when the size of the packet is less than 255 octets (like this tcpdump also has)
- differences in bgp attribute length vs attribute length computed from number of ASPaths (there was this to fix this [https://github.com/deepfield/bird2/pull/3](https://github.com/deepfield/bird2/pull/3))